lib/commands/context.js

```diff
@@ -318,12 +296,16 @@ const extensions = {
       const oldContext = this.curContext;
       this.curContext = `${appIdKey}.${newPage}`;
       // do not wait, as this can take a long time, and the response is not necessary
-      // eslint-disable-next-line promise/prefer-await-to-callbacks, promise/prefer-await-to-then
-      this.remote.selectPage(appIdKey, parseInt(newPage, 10)).catch((err) => {
-        this.log.warn(`Failed to select page: ${err.message}`);
-        this.curContext = oldContext;
-      });
-      this.selectingNewPage = false;
+      (async () => {
+        try {
+          await /** @type {RemoteDebugger} */ (this.remote).selectPage(appIdKey, parseInt(newPage, 10));
```

Comment:
ah, potentially what i saw in the past error about context switch failure (to get a page in webview) was this await related I guess...